### ISSUE TITLE
SA1611 not logging warning, info or error on build

### ISSUE THREAD

**Issue by GenZodd** [DotNetAnalyzers/StyleCopAnalyzers#2218]
I have the following code:

```csharp
/// <summary>
/// Initializes a new instance of the <see cref="HomeController" /> class.
/// </summary>
/// <param name="mapper">The mapper.</param>
public HomeController(IMappingEngine mapper, UserTaskController userTaskController)
{
    this.mapper = mapper;
    this.userTaskController = userTaskController;
}
```

Visual Studio gives me an underline saying `userTaskController` is missing an XML comment. The ruleset is set to error on that but it does not. I have tried warning and info as well and don't see anything logged when I build the project. 

Same issue with SA1614.

**Comment by bjornhellander** [comment]
Hello! I get SA1611 (and SA1614) reported when I select "Build + IntelliSense" in the Error List view, but not if I select "Build Only" instead. Do you see the same behaviour?

**Comment by GenZodd** [comment]
Yes, I see the same thing. But the build is successful even though it is trying to generate an error for this issue.

**Comment by bjornhellander** [comment]
@GenZodd I'm trying to debug this and I'm seeing some strange behaviour. Sometimes... The pattern is not entirely clear to me yet, but I have one question for you: If you open the Properties view for the project that you are having this problem in, do you have "XML documentation file" checked or unchecked in the Build tab?

**Comment by GenZodd** [comment]
@bjornhellander I have "XML documentation file" unchecked. However, when I select that I do get the build to fail and show errors.

**Comment by sharwell** [comment]
@GenZodd What version of StyleCop Analyzers are you using?

**Comment by GenZodd** [comment]
@sharwell I am on 1.0.0. In my version there is no SA0001. Seems odd to me that I have to have XML document file generation on for all documentation rules to function. While they are for sure related, at least to me, they don't seem like they should depend on each other.

**Comment by sharwell** [comment]
@GenZodd In your version SA1652 performs the same function as the new SA0001, but with a different message.

**Comment by GenZodd** [comment]
So it sounds we are saying you have to have XML document file generation on for a number of the rules to work. Still does not seem like they should depend on that to me but ok.

**Comment by sharwell** [comment]
@GenZodd That is a Roslyn limitation. We filed dotnet/roslyn#2773 covering one symptom of this.
